### ISSUE TITLE
Error msg: Connection refused 

### ISSUE THREAD

**Issue by einsteinfreak** [IndrekHaav/photoprism-debian#24]
First: Thanks for this great guide. I'm new to Linux and PhotoPrism but was able to understand and proceed the steps that you described. Good work :-) 

Unluckily I'm not able to login to the PhotoPrism interface yet. 
Using `$ systemctl status photoprism` for analyzes, I get the following information:

`systemd[1]: Starting photoprism.service - PhotoPrism Service...`
`photoprism[738]: level=error msg=dial tcp 192.168.178.38:3306: connection refused"`
`photoprism[1284]: level=error msg=dial tcp 192.168.178.38:3306: connection refused"`
`systemd[1]: photoprism.service: Deactivated successfully.`
`systemd[1]: Started photoprism.service - PhotoPrism Service.`
`systemd[1]: photoprism.service: Consumed 7.282s CPU time. `

I assume that the **server settings for the MariaDB** in the .env-file are not correct. 
The IP of my PC is: `192.168.178.38`
Do you have any advice what I need to change to make it work?

My settings in `/var/lib/photoprism/.env` are: 

`# Database - MariaDB`
`PHOTOPRISM_DATABASE_DRIVER="mysql"`
`PHOTOPRISM_DATABASE_SERVER="192.168.178.38:3306"`
`PHOTOPRISM_DATABASE_NAME="photoprism"`
`PHOTOPRISM_DATABASE_USER="photoprism"`
`PHOTOPRISM_DATABASE_PASSWORD="a_secure_password"`

`# Connection`
`PHOTOPRISM_HTTP_HOST="192.168.178.38"`
`PHOTOPRISM_HTTP_PORT="2342"`

`# Initial password`
`PHOTOPRISM_AUTH_MODE="password"`
`PHOTOPRISM_ADMIN_USER="photoprism"`
`PHOTOPRISM_ADMIN_PASSWORD="a_secure_password"`

`# Storage directories`
`PHOTOPRISM_STORAGE_PATH="/var/lib/photoprism"`
`PHOTOPRISM_ORIGINALS_PATH="/var/lib/photoprism/Originale"`
`PHOTOPRISM_IMPORT_PATH="/var/lib/photoprism/Importe"`

Or is there maybe anything else which needs to be done to make Photoprism run?
Your advice is highly appreciated.

**Comment by IndrekHaav** [comment]
That's a database connection error. You're running PhotoPrism and MariaDB on the same machine? Then try using `127.0.0.1:3306` or `localhost:3306` for the database server address. I believe MariaDB by default doesn't listen on any other addresses.

**Comment by einsteinfreak** [comment]
Thanks for your quick response. 
Yes. PhotoPrism and MariaDB are supposed to run on the same machine. It's my old laptop I'm using for this. Is there anything odd about it? 
My new laptop is supposed to be the client.  

Now, I tried `127.0.0.1:3306`, then `localhost` and finally `127.0.1.1:3306`. After each change of the .env-file I did the daemon-reload and restart and enabling of PhotoPrism. 

The error message always changed accordingly:

127.0.0.1:3306 --> `msg=dial tcp [127.0.0.1]:3306: connection refused`
localhost:3306 --> `msg=dial tcp [::1]:3306: connection refused`
127.0.1.1:3306 --> `msg=dial tcp [127.0.1.1]:3306: connection refused`

Any ideas?

**Comment by IndrekHaav** [comment]
I assume MariaDB is actually running, you have executed `mariadb-secure-installation` and created the PhotoPrism user and database?

Can you connect locally using the `mariadb` CLI utility?

Also see here: https://mariadb.com/kb/en/troubleshooting-connection-issues/

**Comment by einsteinfreak** [comment]
No, I did nothing else than installing Debian 12 (standard setup) and then following all the steps in your PhotoPrism guide. 
Are there any other steps / other requirements that need to be done before I can start PhotoPrism?  If so, can you provide suitable links to me, please?

Forgive me: I'm a Linux noob and do not really understand the questions that you ask me in your last message :-/ But will try to dive into and understand.

**Comment by einsteinfreak** [comment]
I understand that MariaDB needs to be installed first.
Now, I was just following the steps described here: 
https://linuxgenie.net/how-to-install-mariadb-on-debian-12-bookworm-distribution/

Then did the daemon-reload as well as start and enabling of PhotoPrism.
The message 'Connection refused' is gone now. Yeah :+1: 

Got a new one though: `Access denied for user 'photoprism'@'localohost'`
Guess it means that I need to create the PhotoPrism user and database as you said. How do I do this?

**Comment by IndrekHaav** [comment]
Correct, MariaDB needs to be installed separately.

See the PhotoPrism docs for commands to create the database and user: https://docs.photoprism.app/getting-started/advanced/databases/#configuration

**Comment by einsteinfreak** [comment]
BINGO! I finally got to see the PhotoPrism web interface :-)

I followed the instructions in this guide to create the photoprism database and the admin user: http://java.xrheingauerx.de/mariadb_datenbank_anlegen_benutzer_berechtigen.html

Thank you for your support! I think it would be helpful to mention in your guide that prerequisites are:
- Install MariaDB on Debian
- Create the database `photoprism` in MariaDB monitor.
- Create the database `DB User` + permissions in MariaDB monitor.

It might be obvious to advanced users but it certainly is not for beginners.

_By the way: I was studying the documentation on _PhotoPrism.app_ BUT it was not really helpful as it does not say clearly what to do in which order and where._
